### PR TITLE
feat: Expose pii config validation over cabi

### PR DESCRIPTION
**Pull request by untitaker**

### CODE REVIEW

File: py/sentry_relay/processing.py [hunk 1]
```diff
@@ -165,6 +167,21 @@ def is_glob_match(
...
+
+    The parameter is a JSON-encoded string. We should pass the config through
+    as a string such that line numbers from the error message match with what
+    the user typed in.
+    """
+    assert isinstance(config, string_types)
+    try:
+        rustcall(lib.relay_validate_pii_config, encode_str(config))
+    except SerdeJsonError as e:
+        raise ValueError(e.message.split("\n")[0])
```
Comment (jan-auer) [inline py/sentry_relay/processing.py]: Can’t you just throw the original error through?

File: relay-cabi/src/processing.rs [hunk 2]
```diff
@@ -172,6 +172,15 @@ ffi_fn! {
     }
 }
 
+ffi_fn! {
+    unsafe fn relay_validate_pii_config(
```
Comment (jan-auer) [inline relay-cabi/src/processing.rs:177]: Nit: doc comment :)

File: relay-cabi/src/core.rs [hunk 3]
```diff
@@ -60,6 +63,9 @@ impl RelayErrorCode {
                     _ => RelayErrorCode::Unknown,
                 };
             }
+            if let Some(..) = cause.downcast_ref::<serde_json::Error>() {
```
Comment (jan-auer) [inline relay-cabi/src/core.rs]: Nit: Use `is_some` instead?

**Review by jan-auer — APPROVED** [review]

File: py/sentry_relay/processing.py (hunk 1 above)
Comment (untitaker) [reply py/sentry_relay/processing.py]: I would have to split off the useless stacktrace there then

File: relay-cabi/src/core.rs (hunk 3 above)
Comment (untitaker) [reply relay-cabi/src/core.rs]: This is more consistent with the surrounding code

File: relay-cabi/src/processing.rs (hunk 2 above)
Comment (untitaker) [reply relay-cabi/src/processing.rs:177]: Done. I am not sure if ffi_fn will honor it

Comment (jan-auer) [reply relay-cabi/src/processing.rs:177]: It does, see other functions :) The cool thing is that docs are then also in the generated header.

File: py/sentry_relay/processing.py (hunk 1 above)
Comment (jan-auer) [reply py/sentry_relay/processing.py]: I see. In symbolic we do this in the abstraction already. 

It's a little unfortunate that we need to serialize a backtrace and don't just return the message. Yet, we can't return an option nicely via FFI. I'm wondering how you'd call this function from JS in the end?

Comment (untitaker) [reply py/sentry_relay/processing.py]: For now I am only calling it in the API endpoint when persisting, there's already some nice API there to make sure this ends up in the frontend ok

Comment (mitsuhiko) [reply py/sentry_relay/processing.py]: I think there is a potential unicode hazard in there if the message contains data outside ASCII.

**Review by jan-auer — APPROVED** [review]
Looking good. What Armin says still applies, but if you say it's fine I'm okay with accepting the risk.

**Comment by untitaker** [comment]
I think we won't have unicode in the error msg and I don't think this will actually break. In any case this is acceptable risk

**Comment by untitaker** [comment]
Travis being stupid

**Pull request merged**